motr/ut/idx_dix.c

```diff
@@ -804,34 +803,30 @@ static int duc_setup(void)
 							    0x1a);
 	struct m0_fid            srv_dtm0_fid = M0_FID_INIT(0x7300000000000001,
 							    0x1c);
-	struct m0_reqh_service  *cli_srv;
 	struct m0_reqh_service  *srv_srv;
 	struct m0_reqh          *srv_reqh;
 	int                      rc;
 	struct m0_container     *realm = &duc.duc_realm;
 
+	m0_fi_enable("m0_client_init", "hardcoded-dtm0s-fid");
```

Comment:
We already have m0_dtm0_in_ut() function, propose to enable it in addition to hardcoded fid to avoid sleep 15secs during client start. Also may be we can use m0_dtm0_in_ut() instead of hardcoded-dtm0s-fid FI.

Reply:
Unfortunately, `m0_dtm0_in_ut` leads to the situation where we dtm0/service.c has a bit different behavior. `process_clink_cb` is called twice with the same event for the same process:

```c
$4 = (struct m0_fid *) 0x7fffedbb9d90 <dix_ut_sctx+33584>
(gdb) p /x current_proc_fid[0]
$5 = {f_container = 0x7200000000000001, f_key = 0x5}
(gdb) p /x evented_proc_fid[0]
$6 = {f_container = 0x7200000000000001, f_key = 0x6}
(gdb) p process
$7 = (struct dtm0_process *) 0x2b08790
(gdb) p evented_proc_state
$8 = M0_NC_ONLINE
```

It causes the situation where we are trying to initialize the same RPC link twice inside `m0_dtm0_service_process_connect`.

So that I deliberately decided not to use `m0_dtm0_in_ut`, and use a separate FP instead Because of that, the client UT uses explicit `m0_dtm0_service_process_connect` call (see `duc_setup`) to establish the backward connection (server -> client).